client/internal/connect.go

```diff
@@ -68,26 +71,98 @@ func RunClient(ctx context.Context, config *Config) error {
 		if err != nil {
 			return err
 		}
+
+		managementState := nbStatus.ManagementState{
+			URL:       config.ManagementURL.String(),
+			Connected: false,
+		}
+
+		err = statusRecorder.UpdateManagementStatus(managementState)
+		if err != nil {
+			return err
+		}
 		// connect (just a connection, no stream yet) and login to Management Service to get an initial global Wiretrustee config
 		mgmClient, loginResp, err := connectToManagement(engineCtx, config.ManagementURL.Host, myPrivateKey, mgmTlsEnabled,
 			publicSSHKey)
 		if err != nil {
 			log.Debug(err)
-			if s, ok := status.FromError(err); ok && s.Code() == codes.PermissionDenied {
+			if s, ok := gstatus.FromError(err); ok && s.Code() == codes.PermissionDenied {
 				log.Info("peer registration required. Please run `netbird status` for details")
 				state.Set(StatusNeedsLogin)
 				return nil
 			}
 			return wrapErr(err)
 		}
 
+		managementState.Connected = true
+
+		err = statusRecorder.UpdateManagementStatus(managementState)
+		if err != nil {
+			return err
+		}
```

Comment:
I would rather create a new state and update status within a lock of status recorder

Reply:
alternatively create a separate method of statusRecorder that would just update connected state of management, e.g. setManagementConnected(true/false)

Reply:
we can't set the connected status separately as it is related to the URL